retrofit-mock/src/main/java/retrofit2/mock/Calls.java

```diff
@@ -74,7 +74,11 @@ private Calls() {
       if (response != null) {
         return response;
       }
-      throw error;
+      if (error instanceof IOException) {
+        throw (IOException) error;
+      } else {
+        throw new IOException(error);
```

Comment:
I don't think this is what we want because it changes the semantics of the error. If it's a `RuntimeException` subtype we can throw it directly with a cast, if it's an `Error` subtype we can throw it directly with a cast`, and if it's not one of those (and not an `IOException`) then we probably need to [sneaky throw][1] it. And then this behavior needs documented on the factory method.

 [1]: http://www.proofbyexample.com/sneakythrow-avoid-checked-exceptions.html

Reply:
Whoa! "sneaky throw" that's awesome! Done in a98b705.